Ensure that arrays of Classes are handled as strings in ComponentProp…

timothyjward opened the pull request:
…ertyType annotations

Fixes #2698

Signed-off-by: Tim Ward <timothyjward@apache.org>

bjhargrave commented on biz.aQute.bndlib/src/aQute/bnd/component/DSAnnotationReader.java:
```diff
@@ -633,7 +633,9 @@ private void processActivationObject(String propertyDefKey, ClassTypeSignature p
 				Object value = entry.getValue();
-				handleValue(key, value, value instanceof TypeRef, null);
+				boolean isClass = value instanceof TypeRef || (value.getClass()
+					.isArray() && ((Object[]) value)[0] instanceof TypeRef);
```
I think you also need to handle the case where the array is empty. e.g. `@ClassArrayProp({})`. So the `isClass` test should be like:

```
boolean isClass = value instanceof TypeRef || (value.getClass()
	.isArray() && (Array.getLength(value) > 0) && (Array.get(value, 0) instanceof TypeRef));
```

Perhaps another test with an empty array can be added too.

timothyjward commented:
I have added the test and updated the `isClass` calculation. 

@bjhargrave Just to confirm, our expectation is that an empty array will result in no properties in the component xml?

bjhargrave commented:
> @bjhargrave Just to confirm, our expectation is that an empty array will result in no properties in the component xml?

Yes, see [112.8.3](https://osgi.org/specification/osgi.cmpn/7.0.0/service.component.html#service.component-ordering.generated.properties) bullet 2:

> For component property type methods with a value of an empty array, a property element must not be generated.
